src/discord/commands/manage.ts

```diff
@@ -0,0 +1,247 @@
+import { ButtonInteraction, CommandInteraction, MessageEmbed, MessageActionRow, MessageButton, Message, SelectMenuInteraction } from 'discord.js';
+import { Types } from 'mongoose';
+import TimeblockModel from '../../models/timeBlock.model';
+import BookingModel from '../../models/booking.model';
+import SectionModel from '../../models/section.model';
+
+function updateEmbed(userArray: string[], maxCapacity: number, manageState: string) {
+    const embedTitle = manageState === 'add' ? 'Adding' : 'Removing';
+    const preposition = manageState === 'add' ? 'to' : 'from';
+
+    const embed = new MessageEmbed()
+        .setColor('#48d7fb')
+        .setTitle(`Currently ${embedTitle} Collaborators`)
+        .setDescription(
+            `To ${manageState} collaborators ${preposition} your booking, @ their Discord username!\n\nCurrently Invited Collaborators (${
+                maxCapacity! - userArray!.length
+            } available spaces left):\n${userArray.map((user) => `<@!${user}>`).join('\n')}`
+        );
+
+    return embed;
+}
+
+function updateButtons(manageState: string) {
+    const buttonRow = [new MessageButton().setCustomId('completeBooking').setLabel('Complete Booking').setStyle('PRIMARY')];
+    const label = manageState === 'add' ? 'Remove' : 'Add';
+    const style = manageState === 'add' ? 'DANGER' : 'SUCCESS';
+
+    buttonRow.push(new MessageButton().setCustomId(`${label.toLowerCase()}Collaborators`).setLabel(`${label} Collaborators`).setStyle(style));
+
+    return new MessageActionRow().addComponents(buttonRow);
+}
+
+async function mongoDBFilter(userArray: string[], bookingId: string) {
+    //Ensures no duplicate members are listed under users of a certain booking
+    const filterArray = [];
+    const currentUserArray = (await BookingModel.findOne({ _id: Types.ObjectId(bookingId) }))!.users;
+
+    for (const user of userArray) {
+        if (!currentUserArray.includes(user)) {
+            filterArray.push(user);
+        }
+    }
+    return filterArray;
+}
+
+async function parseCommandOptions(interaction: CommandInteraction) {
+    const bookingId = interaction.options.getString('booking-id');
+    //Regex filters out invalid booking ID formats
+    const hexRegex = /[0-9A-Fa-f]{6}/g;
+
+    if (!hexRegex.test(bookingId!)) {
+        interaction.reply({ content: 'Invalid Booking ID Format: Do `/view` to see all your current bookings!', ephemeral: true });
+        return;
+    }
+
+    const bookedBooking = await BookingModel.findOne({ _id: Types.ObjectId(bookingId!) });
+
+    if (bookedBooking === null) {
+        interaction.reply({ content: 'Invalid Booking ID: Do `/view` to see all your current bookings!', ephemeral: true });
+        return;
+    }
+
+    const bookedTimeblock = await TimeblockModel.findOne({ _id: bookedBooking.timeBlock });
+    const bookedSection = await SectionModel.findOne({ _id: bookedTimeblock!.sectionId });
+    const maxCapacity = bookedSection!.capacity;
```

Comment:
Using non-null assertions everywhere isn't very good practice. You should check if they exist first to narrow the typings so that we can actually be confident that they aren't undefined (and properly respond if they are).